tests/unit/test_initial.py

```diff
@@ -38,13 +37,65 @@ def test_list_path_v1(self):
                 self.hvacclient,
                 path='',
                 kv_version=1,
-                source_mount='secret',
-                kv_list=['test']
+                source_mount='secret'
             )
 
-        assert kv_list == ['test', 'test2']
+        assert kv_list == ['test2']
         assert self.hvacclient.secrets.kv.v1.list_secrets.called
 
+    @mock.patch('kv_recursive.list_path')
+    @mock.patch('kv_recursive.recursive_path_builder')
+    def test_list_recursive(self, mock_recursive_path_builder, mock_list_path):
+        kv_recursive.list_recursive(
+            self.hvacclient,
+            path=['test'],
+            kv_version=2,
+            source_mount='secret'
+        )
+
+        assert mock_list_path.called
+        assert mock_recursive_path_builder.called
+
+    @mock.patch('kv_recursive.list_recursive')
+    @mock.patch('kv_recursive.delete_secrets_from_list')
+    def test_delete_recursive(self, mock_delete_secrets_from_list, mock_list_recursive):
+        kv_recursive.delete_recursive(
+            self.hvacclient,
+            path='',
+            kv_version=1,
+            source_mount='secret'
+        )
+
+        assert mock_delete_secrets_from_list.called
+        assert mock_list_recursive.called
+
+    @mock.patch('kv_recursive.list_recursive')
+    @mock.patch('kv_recursive.read_secrets_from_list')
+    def test_read_recursive(self, mock_read_secrets_from_list, mock_list_recursive):
+        kv_recursive.read_recursive(
+            self.hvacclient,
+            path='',
+            kv_version=1,
+            source_mount='secret'
+        )
+
+        assert mock_read_secrets_from_list.called
+        assert mock_list_recursive.called
+
+    @mock.patch('kv_recursive.read_recursive')
+    @mock.patch('kv_recursive.write_secrets_from_list')
+    def test_migrate_secrets(self, mock_write_secrets_from_list, mock_read_recursive):
+        kv_recursive.migrate_secrets(
+            self.hvacclient,
+            self.hvacclient,
+            src_path='',
+            source_mount='secret',
+            dest_mount='secret'
+        )
+
+        assert mock_read_recursive.called
```

Comment:
What do you think about leverage [assert_called_with](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with) here? That way for each of these we know were reading from `source_mount` and writing to `dest_mount`